challenge-16/challenge-16.js

```diff
@@ -1,70 +1,113 @@
-/*
-1. Envolva todo o conteúdo desse desafio em uma IIFE.
-2. Adicione a diretiva 'use strict';
-3. Crie um arquivo index.html e adicione esse script à ele.
-*/
+(function(){
+	'use strict';
+	/*
+	1. Envolva todo o conteúdo desse desafio em uma IIFE.
+	2. Adicione a diretiva 'use strict';
+	3. Crie um arquivo index.html e adicione esse script à ele.
+	*/
 
-/*
-Declare uma variável chamada `name` que receba seu primeiro nome.
-Mostre no console todas as letras do seu nome separadas, com a frase:
-- "[LETRA] é a [POSIÇÃO]ª letra do meu nome."
-Ex: no caso do nome ser "Fernando", deve mostrar as frases:
-- "F é a 1ª letra do meu nome."
-- "e é a 2ª letra do meu nome."
-E assim por diante, até a última.
-*/
-console.log( 'As letras do seu nome:' );
-// ?
+	/*
+	Declare uma variável chamada `name` que receba seu primeiro nome.
+	Mostre no console todas as letras do seu nome separadas, com a frase:
+	- "[LETRA] é a [POSIÇÃO]ª letra do meu nome."
+	Ex: no caso do nome ser "Fernando", deve mostrar as frases:
+	- "F é a 1ª letra do meu nome."
+	- "e é a 2ª letra do meu nome."
+	E assim por diante, até a última.
+	*/
+	console.log( 'As letras do seu nome:' );
+	var name = 'Heverton';
+	//QUAL SERIA A MELHOR SOLUÇÃO DAS 3?
```

Comment:
Na real, se você quiser realmente medir performance, precisaria rodar os 3 códigos uma quantidade grande de vezes, em uma mesma máquina, e tirar uma média.

Não acredito que qualquer um dos 3 vá ter uma diferença muito grande de performance.

Mas, sempre ao escrever um código, não leve em conta só performance. Tem muitas outras coisas que também são importantes se preocupar, como legibilidade, responsabilidades, etc.

Se o seu código está legível, fácil de entender (para outras pessoas, não só pra você), a partir daí você pode começar a se preocupar com performance.

Mas, a menos que a aplicação que você for trabalhar precise de diferenças de microssegundos para execução, a forma como você usa não vai ser um grande problema =)

Escolha sempre a forma que fica mais claro, que gera menos efeitos colaterais (que automaticamente causa menos bugs).